master/internal/task/allocation.go

```diff
@@ -367,13 +377,20 @@ func (a *Allocation) RequestResources(ctx *actor.Context) error {
 // Cleanup ensures an allocation is properly closed. It tries to do everything before failing and
 // ensures we don't leave any resources running.
 func (a *Allocation) Cleanup(ctx *actor.Context) {
-	// This message must be sent when the actor is closing since it closes all
-	// websockets listening for these events.
-	exitReason := okExitMessage
-	if a.exitReason != nil {
-		exitReason = a.exitReason.Error()
+	// These messages must be sent when the actor is closing since it
+	// closes all websockets listening for these events.
+	exitReasons := strings.Join(a.exitReasons, ", ")
+	if exitReasons != "" {
+		a.logger.Insert(ctx, a.enrichLog(model.TaskLog{
+			Log:   "Ended due to " + exitReasons,
```

Comment:
since this isn't going to be a beautiful sentence anyway with the string just, maybe just "exit reasons: %v"